### PR TITLE
refactor(pre-commit): use actions instead of scripts

### PR DESCRIPTION
**Pull request by TomohitoAndo**
Signed-off-by: Tomohito Ando <tomohito.ando@tier4.jp>

## Description

Used [pre-commit action](https://github.com/pre-commit/action) and [git-auto-commit action](https://github.com/stefanzweifel/git-auto-commit-action) instead of running scripts.

related PR

- https://github.com/autowarefoundation/autoware-github-actions/pull/197

## Pre-review checklist for the PR author

The PR author **must** check the checkboxes below when creating the PR.

- [x] I've confirmed the [contribution guidelines].
- [x] The PR follows the [pull request guidelines].

## In-review checklist for the PR reviewers

The PR reviewers **must** check the checkboxes below before approval.

- [x] The PR follows the [pull request guidelines].

## Post-review checklist for the PR author

The PR author **must** check the checkboxes below before merging.

- [x] There are no open discussions or they are tracked via tickets.

After all checkboxes are checked, anyone who has write access can merge the PR.

[contribution guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/
[pull request guidelines]: https://autowarefoundation.github.io/autoware-documentation/main/contributing/pull-request-guidelines/

### CODE REVIEW

**Comment by TomohitoAndo** [comment]
I confirmed this works.
https://github.com/tier4/gsm8_interface/actions/runs/3584226014/jobs/6030625061
(TIER IV internal link)

**Comment by kenji-miyake** [comment]
@TomohitoAndo It's okay for me, but can I ask the reason (your motivation)?

**Comment by kenji-miyake** [comment]
@TomohitoAndo Also, could you write your previous PR in the description, please? :pray:

**Comment by TomohitoAndo** [comment]
@kenji-miyake 
The reason is that I think this is more clean and easy to read.
But I don't know much about github actions, I'd like your opinion :pray:

**Comment by kenji-miyake** [comment]
@TomohitoAndo I see. In that case, using GitHub Actions to make it simple is recommended in most time.

File: pre-commit/action.yaml [hunk 1]
```diff
@@ -18,22 +18,14 @@ runs:
...
-      run: |
-        if ! git diff --quiet; then
-          git add -u
-          git commit -m "pre-commit fixes"
-          git push
-        fi
-      shell: bash
+      uses: stefanzweifel/git-auto-commit-action@v4
+      with:
+        commit_message: "ci(pre-commit): autofix"
```
Comment (kenji-miyake) [inline pre-commit/action.yaml:31]: Maybe using `style` is better?
https://github.com/commitizen/conventional-commit-types/blob/d1fb9cc953b8f1e33e2dd63babdf6ae8ab1c8711/index.json#L15-L18

Or using `pre-commit fixes` to be the same as `pre-commit.ci` is acceptable.

```suggestion
        commit_message: "style(pre-commit): autofix"
```

Comment (TomohitoAndo) [reply pre-commit/action.yaml:31]: Thanks! I also think that is better.
But I referred this commit message from `pre-commit.cii`.
image
https://github.com/autowarefoundation/autoware.universe/pull/2333/commits/6eee79d4358add752ada35316eb72b774fb24dab

Comment (kenji-miyake) [reply pre-commit/action.yaml:31]: Ah, I see. It's from https://github.com/autowarefoundation/autoware.universe/blob/8bbc5052588c7909e4fa52b6fdbcd6678f1afc75/.pre-commit-config.yaml#L2.

Maybe I made a mistake. I'll check why I used this message.

Comment (kenji-miyake) [reply pre-commit/action.yaml:31]: It was modified from the example in https://pre-commit.ci/.
image

Let's change `.pre-commit-config.yaml` as well!

Comment (TomohitoAndo) [reply pre-commit/action.yaml]: I fixed :smile: 
https://github.com/autowarefoundation/autoware-github-actions/pull/198/commits/04b6773d2836bc192610a3ef919c8f5deb77527e

Comment (kenji-miyake) [reply pre-commit/action.yaml]: Thanks. For other repositories, I'll send PRs.

**Review by kenji-miyake — APPROVED** [review]

File: pre-commit/action.yaml (hunk 1 above)
Comment (kenji-miyake) [reply pre-commit/action.yaml]: https://github.com/autowarefoundation/autoware/pull/3094

**Pull request merged**